datanode/gateway/graphql/schema.graphql

```diff
@@ -4088,6 +4088,22 @@ enum ProposalRejectionReason {
   PROPOSAL_ERROR_INVALID_PERPETUAL_PRODUCT
   "Spot market proposal contains too many size decimal places"
   PROPOSAL_ERROR_INVALID_SIZE_DECIMAL_PLACES
+  "LP price range must be larger than 0"
+  PROPOSAL_ERROR_LP_PRICE_RANGE_NONPOSITIVE
+  "LP price range must not be larger than 100"
+  PROPOSAL_ERROR_LP_PRICE_RANGE_TOO_LARGE
+  "Linear slippage factor is out of range, either negative or too large"
+  PROPOSAL_ERROR_LINEAR_SLIPPAGE_FACTOR_OUT_OF_RANGE
+  "Quadratic slippage factor is out of range, either negative or too large"
+  PROPOSAL_ERROR_QUADRATIC_SLIPPAGE_FACTOR_OUT_OF_RANGE
+  "Referral program proposal is invalid"
+  PROPOSAL_ERROR_INVALID_REFERRAL_PROGRAM
+  "Volume discount program proposal is invalid"
+  PROPOSAL_ERROR_INVALID_VOLUME_DISCOUNT_PROGRAM
+  "One or more proposals in a batch has been rejected"
+  PROPOSAL_ERROR_PROPOSAL_IN_BATCH_REJECTED
+  "One or more proposals in a batch has been declined"
```

Comment:
I believe the rejection could be for a number of reasons including proposals that are malformed for example, so if you have say one proposal in a batch of X that failed validation, it could get rejected.

It's the same text that's on the proto documentation anyway, so I guess we should be fine, otherwise we should change both.

Reply:
@guoguojin - I meant the "declined" error rather than rejected, I think rejected is clear. But as it's just replicating what's already in protos, I take your point :)